FWCore/Modules/src/TimeStudyModules.cc

```diff
@@ -261,6 +262,7 @@ namespace timestudy {
         }
       }
       waitingTaskPerStream_.clear();
+      waitingTaskPerStream_.resize(waitingTaskPerStream_.capacity());
```

Comment:
At this stage the thread is about to end, so it is not clear to me why the resizing the `waitingTaskPerStream_` would help. Nothing should access the `waitingTaskPerStream_` at this point.

Or is the idea to make the `asyncWork()` to not write to invalid memory? In that case, how about changing the `[]` to `at()` there?

Reply:
The way I think about it, the vector capacities should be an invariant, and the size to where possible.  Multi-thread access is easier to reason about if you don't have to worry about iterator invalidation, though it's true this one doesn't really matter.